internal/controller/factory/statefulset.go

```diff
@@ -482,3 +374,269 @@ func mergeWithDefaultProbe(probe *corev1.Probe, defaultProbe corev1.Probe) *core
 func hasProbeHandlerAction(probe corev1.Probe) bool {
 	return probe.HTTPGet != nil || probe.TCPSocket != nil || probe.Exec != nil || probe.GRPC != nil
 }
+
+func (c certificate) Name() string {
+	return c.certName
+}
+
+func (c certificate) MountPath() string {
+	return c.mountPath
+}
+
+func (c certificate) CrtFilePath() string {
+	return c.buildFilePath(c.crtFile)
+}
+
+func (c certificate) KeyFilePath() string {
+	if c.keyFile == "" {
+		return ""
+	}
+	return c.buildFilePath(c.keyFile)
+}
+
+func (c certificate) buildFilePath(fileName string) string {
+	return filepath.Join(c.mountPath, fileName)
+}
+
+func getCertificateConfig(name, mountPath, crtFile, keyFile string) certificate {
+	return certificate{
+		certName:  name,
+		mountPath: mountPath,
+		crtFile:   crtFile,
+		keyFile:   keyFile,
+	}
+}
+
+func (e etcd) ContainerName() string {
+	return e.containerName
+}
+
+func (e etcd) VolumeName() string {
+	return e.volumeName
+}
+
+func (e etcd) MountPath() string {
+	return e.mountPath
+}
+
+func (e etcd) DirName() string {
+	return e.dirName
+}
+
+func (e etcd) DataPath() string {
+	return filepath.Join(e.mountPath, e.dirName)
+}
```

Comment:
As a follow up to my previous comment, I would drop this for now and, rather than having functions, that accept an `etcdConfigurator` interface, I'd write functions that accept an `etcd` struct.

Compare

```go
func someFunc(ec etcdConfigurator) {
        doSomethingWith(ec.VolumeName())
}
```

and

```go
func someFunc(e etcd) {
        doSomethingWith(e.volumeName)
}
```